agent/src/beerocks/slave/beerocks_slave_main.cpp

```diff
@@ -505,6 +514,9 @@ int main(int argc, char *argv[])
         }
     }
 
+    //Handle -n option (non-intel agent)
+    beerocks_slave_conf.non_intel = g_non_intel;
```

Comment:
I find it weird to save execution flag where the configuration from the configuration files are.
IMO, adding it to the constructor of the slave is the right way to go.
Maybe even add a struct of execution flags to the constructor.
Another option is to add this field on the config file instead of using flag, and whoever runs the agent, will change the config file field value, before running the agent.

Reply:
Moran - please remove your review. We will merge this as is since it blocks our progress in passing 4.2.1 testcase, the rest of the comments will be discussed in architecture calls and fixed if needed in a different PR.